test/parallel/test-http2-max-invalid-frames.js

```diff
@@ -23,7 +23,10 @@ server.on('stream', (stream) => {
 
 server.listen(0, () => {
   const h2header = Buffer.alloc(9);
-  const conn = net.connect(server.address().port);
+  const conn = net.connect({
+    port: server.address().port,
+    allowHalfOpen: true
+  });
```

Comment:
@lundibundi Could you take a look here? The readable side of the connection cleanly ends, cleanups the conn and thus never errors. Fix is to not destroy on `'end'`. Does that somehow invalidate the test?

Reply:
Related to: https://github.com/nodejs/node/issues/30505

Reply:
@ronag Sorry for the late response, I missed this one. I'm not sure I understand the reasons for this 
> The readable side of the connection cleanly ends, cleanups the conn and thus never errors

The server connection handler shouldn't be called as we are only sending 'close stream' packets (and never open a stream which is therefore invalid) and we don't `.end` \ `.destroy` the connection on the client-side. Something doesn't feel right here, not sure though.

As for the test itself, I think it will be fine either way and shouldn't be affected.

/cc @addaleax as the author of the original test (test-http2-reset-flood)

Reply:
I’m not sure I understand why this is necessary either, but I agree that the test functionality shouldn’t be affected – the important thing is that the `.once('error')` handler below fires.